test/controllers/api/v2/host_tracer_controller_test.rb

```diff
@@ -28,53 +22,25 @@ def test_index
       assert_includes results['results'].collect { |item| item['id'] }, @host1.host_traces.first.id
     end
 
-    def test_resolve_group_by_helper
+    def test_resolve
       trace_one = Katello::HostTracer.create(host_id: @host1.id, application: 'rsyslog', app_type: 'daemon', helper: 'systemctl restart rsyslog')
-      trace_two = Katello::HostTracer.create(host_id: @host2.id, application: 'rsyslog', app_type: 'daemon', helper: 'systemctl restart rsyslog')
-      helper = {:helper => trace_two.helper}
-
-      job_invocation = {"description" => "Restart Services", "id" => 1, "job_category" => "Katello"}
-      JobInvocationComposer.expects(:for_feature).with(:katello_service_restart, [@host1.id, @host2.id], helper).returns(mock(trigger: true, job_invocation: job_invocation))
+      task_output = {"id" => nil, "label" => nil, "started_at" => nil, "ended_at" => nil, "state" => nil, "result" => nil, "external_id" => nil,
```

Comment:
It looks like this works but imho it's purely a coincidence that this output is matching with the below `ForemanTasks::Task.new`. I recommend using the same task in both places. ie

```
task = ForemanTasks::Task.new
....

Which you can return from the stubbed find method, and you can call `to_json` on it to make your assertion in the response body

Reply:
Updated